src/workers/event.rs

```diff
@@ -0,0 +1,545 @@
+            .await?
+        {
+            trace!(actor = ?current, "Found existing actor data");
+            // Merge in current counts
+            actor_data.count = current.count;
+        }
+
+        // Update count of the data
+        actor_data
+            .count
+            .entry(actor.host_id.clone())
+            .and_modify(|val| *val += 1)
+            .or_insert(1);
+
+        self.store
+            .store(lattice_id, actor.public_key.clone(), actor_data)
+            .await
+            .map_err(anyhow::Error::from)
+    }
+
+    #[instrument(level = "debug", skip(self, actor), fields(actor_id = %actor.public_key, host_id = %actor.host_id))]
+    async fn handle_actor_stopped(
+        &self,
+        lattice_id: &str,
+        actor: &ActorStopped,
+    ) -> anyhow::Result<()> {
+        trace!("Removing stopped actor from store");
+        debug!("Fetching current data for actor");
+        if let Some(mut current) = self
+            .store
+            .get::<Actor>(lattice_id, &actor.public_key)
+            .await?
+        {
+            trace!(actor = ?current, "Found existing actor data");
+            if let Some(current_count) = current.count.get(&actor.host_id) {
+                let new_count = current_count - 1;
+                if new_count == 0 {
+                    trace!(host_id = %actor.host_id, "Stopped last actor on host, removing host entry from actor");
+                    current.count.remove(&actor.host_id);
+                } else {
+                    trace!(count = %new_count, host_id = %actor.host_id, "Setting current actor");
+                    current.count.insert(actor.host_id.clone(), new_count);
+                }
+            }
+
+            if current.count.is_empty() {
+                debug!("Last actor instance was removed, removing actor from storage");
+                self.store
+                    .delete::<Actor>(lattice_id, &actor.public_key)
+                    .await
+            } else {
+                self.store
+                    .store(lattice_id, actor.public_key.clone(), current)
+                    .await
+            }?;
+        }
+        Ok(())
+    }
+
+    #[instrument(level = "debug", skip(self, host), fields(host_id = %host.id))]
+    async fn handle_host_heartbeat(
+        &self,
+        lattice_id: &str,
+        host: &HostHeartbeat,
+    ) -> anyhow::Result<()> {
+        debug!("Updating store with current host heartbeat information");
+        // Host updates just overwrite current information, so no need to fetch
+        let host_data = Host::from(host);
+        self.store
+            .store(lattice_id, host.id.clone(), host_data)
+            .await?;
+
+        // NOTE: We can return an error here and then nack because we'll just reupdate the host data
+        // with the exact same host heartbeat entry. There is no possibility of a duplicate
+        self.heartbeat_provider_update(lattice_id, host).await?;
```

Comment:
Nope, it just results in a redelivery. Technically any of the events can do this, but my comment here was noting that we won't end up with duplicate counts by returning an error here (because we are just overwriting data). By default, it takes 3 message redeliveries to have NATS poison pill us